### PR TITLE
[DEVHUB-1074] Setup Kanopy and Drone Pipeline

### PR DESCRIPTION
**Pull request by ajhenri**
This PR introduces a basic `Dockerfile`, Kanopy environment configuration, and Drone pipeline for the Next.js builds and deployments (staging and production).

The build steps are the same for both staging and production, except they use a different Strapi source, which is necessary since the content is pulled on each build.
 
Summary of changes:

- `Dockerfile` and .dockerignore
- Drone pipeline for automatic staging deployment. The staging application will be deployed on each push to the `main` branch.
- On Drone promotion to production target, the application will be rebuilt with the Strapi production content.
- Added configuration for health probe, k8s resources, and autoscaling, based on the Kanopy production checklist.

Staging Deploy Example: https://drone.corp.mongodb.com/mongodb/devcenter/51
Production Deploy Example: https://drone.corp.mongodb.com/mongodb/devcenter/52

### CODE REVIEW

File: next.config.js [hunk 1]
```diff
@@ -3,6 +3,14 @@ const configVals = {
...
         domains: ['mongodb-devhub-cms.s3.us-west-1.amazonaws.com'],
     },
+    rewrites: async () => {
+        return [
+            {
+                source: '/healthz',
+                destination: '/api/health',
+            },
+        ];
+    },
```
[the remark this answers is not in the document]
Comment (harikakotipalli) [reply next.config.js:13]: is this redirect for exposing /healthz as /api/health - the reasoning for this is Kanopy not exposing all paths ?

[the remark this answers is not in the document]
Comment (harikakotipalli) [reply next.config.js:13]: This is very cool - if we can append /developer to all the paths.

**Review by harikakotipalli — APPROVED** [review]
Great Job @ajhenri. Very quick turn around - appreciate it.

File: next.config.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (ajhenri) [reply next.config.js:13]: > is this redirect for exposing /healthz as /api/health - the reasoning for this is Kanopy not exposing all paths ?

Good question, Kanopy exposes `api/health` so I could've used that as well. I made the rewrite for a cosmetic reason; to be consistent with both Kanopy and [Kubernetes documentation](https://kubernetes.io/docs/tasks/configure-pod-container/configure-liveness-readiness-startup-probes/#define-a-liveness-http-request) and use the standard liveness probe path `/healthz`. I also wanted to use rewrites with Next.js to set up a foundation here for the future.

**Comment by ajhenri** [comment]
Thanks, y'all! going to merge this in

**Pull request merged**